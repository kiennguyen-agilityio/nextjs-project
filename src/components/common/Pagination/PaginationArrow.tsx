import Link from 'next/link';

// icons
import { ChevronRightIcon, ChevronLeftIcon } from '@/icons';

// components
import { Button } from '@/components/common/Button';

interface PaginationArrowProps {
  href: string;
  direction: 'left' | 'right';
  isDisabled?: boolean;
}

const PaginationArrow = ({
  href,
  direction,
  isDisabled,
}: PaginationArrowProps) => {
  const baseClass =
    'flex h-8 w-8 items-center justify-center rounded border-none';
  const disabledClass = isDisabled
    ? 'pointer-events-none text-gray-300'
    : 'hover:bg-gray-200';
  const marginClass = direction === 'left' ? 'mr-2' : 'ml-2';

  const className = `${baseClass} ${disabledClass} ${marginClass}`;

  const icon =
    direction === 'left' ? <ChevronLeftIcon /> : <ChevronRightIcon />;

  return isDisabled ? (
    <Button customClass={className} aria-disabled="true" variant="outline">
      {icon}
    </Button>
  ) : (
    <Link href={href} className={className}>
      {icon}
    </Link>
  );
};

export default PaginationArrow;
