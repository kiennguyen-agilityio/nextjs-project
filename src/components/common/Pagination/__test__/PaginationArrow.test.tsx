import { render } from '@testing-library/react';
import PaginationArrow from '../PaginationArrow';

describe('PaginationArrow', () => {
  test('should render PaginationArrow component', () => {
    const { asFragment } = render(
      <PaginationArrow href="/page/2" direction="left" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
