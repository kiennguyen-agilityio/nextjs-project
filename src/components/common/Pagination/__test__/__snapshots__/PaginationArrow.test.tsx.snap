// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationArrow should render PaginationArrow component 1`] = `
<DocumentFragment>
  <a
    class="flex h-8 w-8 items-center justify-center rounded border-none hover:bg-gray-200 mr-2"
    href="/page/2"
  >
    <svg
      class="w-4 h-4"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.75 19.5 8.25 12l7.5-7.5"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </a>
</DocumentFragment>
`;
